test/persistence-hooks.suite.js

```diff
@@ -1150,6 +1150,41 @@ module.exports = function(dataSource, should) {
         });
       });
 
+      it('applies updates from `persist` hook - for nested model instance', function(done) {
+        var Address = dataSource.createModel('Address', {
+          city: { type: String, required: true },
+          country: { type: String, required: true }
+        });
+
+        var User = dataSource.createModel('User', {
+          id: { type: String, id: true, default: uid },
+          name: { type: String, required: true },
+          address: {type: Address, required: false}
+        });
+
+        User.create({name: 'Joe'}, function(err, instance) {
+          if (err) return done(err);
+
+          var existingUser = instance;
+
+          User.observe('persist', pushContextAndNext(function(ctx){
+            ctx.data.extra = 'hook data';
```

Comment:
Please add an assertion that `ctx.data.address` is set and a plain object (not an instance of Address model).